### ISSUE TITLE
[Cpg] extra Cpg Node to represent the argument of function/method call

### ISSUE THREAD

**Issue by d1tto** [joernio/joern#3907]
**Is your feature request related to a problem? Please describe.**

The PHP language has properties related to the arguments of a function call, such as array unpacking: `test(...$arg)`. This marks the argument `$arg` as `unpack` in the AST. There is no such property in the CPG. Other programming languages have similar properties, such as python, javascript

**Describe the solution you'd like**

Add an additional node in CPG to represent the function arguments. It can be just a wrapper of other nodes.

**Describe alternatives you've considered**

It also can be represented as a string in the code field of the function call node. For example: `test(...$arg)`，in current CPG，the `code` filed of `Call` node is `test($arg)` instead of `test(...$arg)`

**Comment by DavidBakerEffendi** [comment]
Ah, I see this is on the corresponding parameter node. So `call.argument.parameter.isVariadic`. One would need a working call graph here, but I haven't investigated the call graph in the php frontend yet.

**Comment by DavidBakerEffendi** [comment]
Yeah, I see the value in that. It may be valuable to have a similar modifier here to know when the unpacking happens at the call site.

**Comment by d1tto** [comment]
This probably requires the following changes
- add a new modifier named "UNPACK" to `ModifierTypes` in `codepropertygraph` domains.
- add UNPACK modifier as the children AST to argument if the argument is unpacked in PHP (and other language) frontend like below pseudo code
  ```
  val arguments = call.args.map(astForCallArg)
  arguments.map(arg => if (arg.isUnpack) arg.withChildren(Modifier.UNPACK) else arg)
  ```

is right?

**Comment by DavidBakerEffendi** [comment]
It is unclear if we'd want this to be a property or a modifier, since it's a schema change it is an API change, so I would want to involve others in the conversation.

@maltek @max-leuthaeuser @ml86 @fabsx00

**Comment by d1tto** [comment]
> It is unclear if we'd want this to be a property or a modifier, since it's a schema change it is an API change, so I would want to involve others in the conversation.
> 
> @maltek @max-leuthaeuser @ml86 @fabsx00

Thanks. 
I think it is necessary to support it for the more precise data flow analysis of joern itself and the secondary developers based on joern

**Comment by d1tto** [comment]
I think https://github.com/joernio/joern/pull/3908 is a tmp solution, which make the `code` field is more correct without change the Cpg schema and api. I can parse the code field to identify which argument is unpacked. 
Could you help me review it? Thanks.

**Comment by kar3nk** [comment]
So right now, is there a way to correctly parse unpack in python like **args?
